src/lib/support/SerializableIntegerSet.h

```diff
@@ -0,0 +1,141 @@
+
+    /**
+     * @brief
+     *   Serialize the sparse array into a base 64 encoded string.
+     *   Only the values till mNextAvailable index are encoded.
+     *   The empty indexes between 0, and mNextAvailable, are also
+     *   encoded.
+     *
+     * @param[in] buf Buffer where serialized string is written
+     * @param[in] buflen Length of buf
+     * @return pointer to buf, or nullptr in case of error
+     */
+    const char * SerializeBase64(char * buf, uint16_t & buflen);
+
+    /**
+     * @brief
+     *   Deserialize a base64 encoded string into the sparse array.
+     *   The mNextAvailable index is calculated based on how many
+     *   values are in the deserialized array.
+     *
+     * @param[in] serialized Serialized buffer
+     * @param[in] buflen Length of buffer
+     * @return CHIP_NO_ERROR in case of success, or the error code
+     */
+    CHIP_ERROR DeserializeBase64(const char * serialized, uint16_t buflen);
+
+    /**
+     * @brief
+     *   Get the length of string if the array is serialized.
+     */
+    uint16_t SerializedSize() { return static_cast<uint16_t>(BASE64_ENCODED_LEN(sizeof(uint64_t) * mNextAvailable)); }
+
+    /**
+     * @brief
+     *   Get the maximum length of string if the array were full and serialized.
+     */
+    uint16_t MaxSerializedSize() { return static_cast<uint16_t>(BASE64_ENCODED_LEN(sizeof(uint64_t) * mCapacity)); }
+
+    /**
+     * @brief
+     *   Check if the value is in the array.
+     *
+     * @param[in] value Value to find
+     * @return True, if it's prsent in the array.
+     */
+    bool Contains(uint64_t value) { return Find(value) != mCapacity; }
+
+    /**
+     * @brief
+     *   Insert the value in the array. If the value is duplicate, it
+     *   won't be inserted.
+     *
+     * @return CHIP_NO_ERROR in case of success, or the error code
+     */
+    CHIP_ERROR Insert(uint64_t value);
+
+    /**
+     * @brief
+     *   Delete the value from the array.
+     */
+    void Remove(uint64_t value);
+
+private:
+    uint64_t * const mData;
+    const uint16_t mCapacity;
+    uint16_t mNextAvailable;
+
+    uint16_t FirstAvailableForUniqueId(uint64_t value);
+
+    /**
+     * @brief
+     *   Find a value in the array.
+     *
+     * @param[in] value Value to find
+     * @return index of the value if found, or max length (mCapacity) of the array
+     */
+    uint16_t Find(uint64_t value);
+
+    uint64_t ToLE64(uint64_t value) { return Encoding::LittleEndian::Get64(reinterpret_cast<const uint8_t *>(&value)); }
+};
+
+template <uint16_t kCapacity>
+class SerializableU64Set : public SerializableU64SetBase
+{
+public:
+    SerializableU64Set() : SerializableU64SetBase(mBuffer, kCapacity)
+    {
+        nlSTATIC_ASSERT_PRINT(kCapacity < UINT16_MAX / sizeof(uint64_t),
+                              "Serializable u64 set capacity cannot be more than UINT16_MAX / sizeof(uint64_t)");
```

Comment:
explaining the restriction would be useful.

I imagine we want to ensure base64 max sizes will fit in uint64_t, however that does not seem immediately obvious.